### PR TITLE
Fix C4 dissapearance on inclined surfaces with Valve HLDS

### PR DESCRIPTION
**Pull request by In-line**
Thanks  to the great master of the ancient magic @s1lentq.
Related to dreamstalker/rehlds#555

### CODE REVIEW

**Comment by Vaqtincha** [comment]
gryazniy xak :P

**Comment by In-line** [comment]
@Vaqtincha Predloji luchshe

File: regamedll/dlls/ggrenade.cpp [hunk 1]
```diff
@@ -1141,10 +1145,21 @@ void AnnounceFlashInterval(float interval, float offset)
 void CGrenade::C4Think()
 {
 	if (!IsInWorld())
+#ifdef REGAMEDLL_FIXES
+	{
+		pev->origin = pev->oldorigin;
+		
+		if(DROP_TO_FLOOR(edict()) > 0)
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/ggrenade.cpp]: Space.

File: regamedll/dlls/ggrenade.cpp [hunk 2]
```diff
@@ -1141,10 +1145,21 @@ void AnnounceFlashInterval(float interval, float offset)
...
+		{
+			pev->velocity = g_vecZero;
+		}
+	}
+#else
 	{
 		UTIL_Remove(this);
 		return;
 	}
+#endif
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/ggrenade.cpp]: Maybe?
```C++
	if (!IsInWorld())
	{
#ifdef REGAMEDLL_FIXES
		pev->origin = pev->oldorigin;
		
		if (DROP_TO_FLOOR(edict()) > 0)
		{
			pev->velocity = g_vecZero;
		}
#else
		UTIL_Remove(this);
		return;
#endif
	}
```

**Comment by In-line** [comment]
@WPMGPRoSToTeMa Thanks.

**Comment by In-line** [comment]
Tested :white_check_mark: , works correctly on last ReHLDS and Valve's HLDS.

**Pull request merged**